client/helpers/scenes.ts

```diff
@@ -187,199 +179,319 @@ export async function createSceneAndSceneItems({
-    client,
-    createSceneItemReqs,
-    failFast,
-    onProgress,
-    parallelism,
-    sceneId,
+  if (verbose) onMsg(`Scene ID: ${sceneId}`);
+  if (verbose) onMsg(`Creating scene items...`);
+  let itemCount = 0;
+  let createFailed = false;
+  let sceneResult;
+
+  const reqMap: Map<string, CreateSceneItemRequest[]> = new Map();
+  // create parent map and set ordinals based on request order
+  createSceneItemReqs.forEach((req) => {
+    const reqParent =
+      req.data.attributes.parent ??
+      req.data.relationships.parent?.data.id ??
+      '';
+    if (!reqMap.has(reqParent)) {
+      reqMap.set(reqParent, []);
+    }
+    const siblings = reqMap.get(reqParent);
+    if (req.data.attributes.ordinal == null) {
+      req.data.attributes.ordinal = siblings?.length;
+    }
+    siblings?.push(req);
   });
-  const { a: queuedItems, b: errors } = partition(
-    createRes.queuedSceneItems,
-    (i) => isQueuedJob(i.res)
-  );
-  const queued = returnQueued ? createRes.queuedSceneItems : [];
-  if (queuedItems.length === 0 || errors.length === createRes.leaves) {
-    return { errors, queued, scene };
+
+  // sort all scene item requests into depth sorted array of arrays
+  const depthSortedItems: CreateSceneItemRequest[][] = [];
+  // fetch list of scene items with no parent (root items)
+  let nextChildren: CreateSceneItemRequest[] = reqMap.get('') || [];
+  reqMap.delete('');
+  while (nextChildren.length > 0) {
+    depthSortedItems.push(nextChildren);
+    nextChildren = nextChildren.flatMap((req) => {
+      if (
+        req.data.attributes.suppliedId &&
+        reqMap.has(req.data.attributes.suppliedId)
+      ) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const children = reqMap.get(req.data.attributes.suppliedId)!;
+        reqMap.delete(req.data.attributes.suppliedId);
+        return children;
+      } else {
+        return [];
+      }
+    });
   }
 
-  if (verbose) onMsg(`Polling for completed scene-items...`);
-
-  const limit = pLimit(Math.min(parallelism, 20));
-  const cnt = queuedItems.length;
-  const step = Math.floor(cnt / ((PollPercentage / 100) * cnt));
-  const qis = [];
-  // Poll for percentage of items starting at end of `queuedItems` array
-  for (let i = 0; i < cnt; i += step) qis.push(queuedItems[cnt - i - 1]);
-
-  async function poll({ req, res }: QueuedSceneItem): Promise<void> {
-    const r = await pollQueuedJob<SceneItem>({
-      id: (res as QueuedJob).data.id,
-      getQueuedJob: (id, cancelToken) =>
-        client.sceneItems.getQueuedSceneItem({ id }, { cancelToken }),
-      allow404: true,
-      limit,
-      polling,
+  let resultQueuedOps: QueuedBatchOps[] = [];
+  let resultBatchErrors: QueuedBatchOps[] = [];
+  let resultItemErrors: SceneItemError[] = [];
+
+  // if we had any scene item requests with invalid parents,
+  // add error entries indicating so.
+  reqMap.forEach((children: CreateSceneItemRequest[]) => {
```

Comment:
This is a bit tricky to follow. Why is `reqMap` now a map of items without parents?